radis/lbl/base.py

```diff
@@ -3239,54 +3240,116 @@ def plot_populations(self, what="vib", isotope=None, nfig=None):
 
 
 def get_waverange(
+    wmin=None,
+    wmax=None,
+    wunit=None,
     wavenum_min=None,
     wavenum_max=None,
     wavelength_min=None,
     wavelength_max=None,
     medium="air",
 ):
-    """ Returns wavenumber based on whatever input was given: either ν_min, ν_max 
-    directly, or λ_min, λ_max  in the given propagation ``medium``.
-
-    Parameters
-    ----------
-
-    medium: ``'air'``, ``'vacuum'``
-        propagation medium
-
-    wavenum_min, wavenum_max: float, or ~astropy.units.quantity.Quantity or ``None``
-        wavenumbers
-
-    wavelength_min, wavelength_max: float, or ~astropy.units.quantity.Quantity or ``None``
-        wavelengths in given ``medium``
-
-    Returns
-    -------
+    represent_wavelength = False
+    represent_wavenum = False
+
+    # user did not pass wunit
+    if isinstance(wunit, Default):
+        wunit = wunit.value
+        # user did not pass wmin or wmax
+        if wmin is None or wmax is None:
+            assert wmin is None and wmax is None
+        else:
+            assert wmin is not None
+            assert wmax is not None
+            # user passes wmin/wmax with unit
+            if isinstance(wmin, u.Quantity) or isinstance(wmax, u.Quantity):
+                assert wmin.unit.is_equivalent(u.m) or wmin.unit.is_equivalent(1 / u.m)
+                assert wmax.unit.is_equivalent(u.m) or wmax.unit.is_equivalent(1 / u.m)
+                assert wmin.unit.is_equivalent(wmax.unit)
+                if wmin.unit.is_equivalent(u.m):
+                    represent_wavelength = True
+                else:
+                    represent_wavenum = True
+            # user did not pass wmin/wmax with unit
+            else:
+                assert not isinstance(wmin, u.Quantity)
+                assert not isinstance(wmax, u.Quantity)
+                wmin = wmin * u.Unit(wunit)
+                wmax = wmax * u.Unit(wunit)
+                represent_wavenum = True
```

Comment:
it works, but are you sure we cannot simplify all these if/else?

Reply:
I think there must be a few simplifications we can make. I was under the impression that keeping the code structure like this would make it easier to follow and debug, especially now that we have quite a few ways of calling the `get_waverange` function.

Reply:
We are starting with an `if` conditional on `wunit`, although it will be irrelevant if wmin, wmax are not given. 

In general, we are mixing (in the new implementation, but the old one too!) unit conversions to base units (nm / cm-1), final conversion to cm-1 (which requires to check the propagation medium: air or vacuum), and tests of input consistency (_anything with assertion tests, or raise errors_). It's a little hard to understand.

What about refactoring to check these sequentially:
```
# 1. Check input consistency
- Check that at least a pair of inputs is given 
- Check that not too types of inputs are given at the same time
- If wavenum or wavelengh, or wmin is dimensionned, check that wunit is consistent (if not default)
- [we can maybe leave the check that Medium is given in the wavelength case for later]

# 2. Conversion to base units (strip dimensioned units, convert to nm or cm )
# Now no need to assert / raise anything. 
wmin, wmax given
 --> if not dimensioned: convert to nm or cm-1 based on wunit  (no need to assert, we did it in 1.). Then write wavelength_min/max , or wavenum_min/max
 --> if dimensioned, convert to nm, or cm-1. Then write wavelength_min/max, or wavenum_min/max
wavenum_min, wavenum_max given
 --> convert to cm-1
wavelength_min, wavelength_max given:
--> convert to nm

# now we only have wavenum_min/max, or wavelen_min/max, in standard units (cm-1, or nm). Last thing to check: converting nm > cm-1 depends on the propagation medium (air/vacuum)
# 3. Convert to cm-1 
if wavenum_min/max: 
--> nothing to do 
if wavelength_min/max
--> check that medium is given
--> convert to cm-1 with nm2cm or nm_air2cm  (handy function in convert.py) 

Return wavenum_min/max
```

Reply:
Got it! I was a little confused and started thinking along the lines of simplifying the code from the logic point of view. I'll start working on this right away!

Reply:
Any suggestion welcome, it's more of a design than a coding job ! :)